python/pyiceberg/io/__init__.py

```diff
@@ -257,6 +257,8 @@ def delete(self, location: Union[str, InputFile, OutputFile]) -> None:
     "gcs": [ARROW_FILE_IO],
     "file": [ARROW_FILE_IO],
     "hdfs": [ARROW_FILE_IO],
+    "abfs": [FSSPEC_FILE_IO],
```

Comment:
Didn't know if PyArrow would also apply here.

Reply:
Let's just include what is tested for now. We can check for pyarrow support and add it along with tests if it's there.